src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -2935,7 +2935,7 @@ void processContextElement
   std::map<std::string, std::string>&  uriParams,   // FIXME P7: we need this to implement "restriction-based" filters
   const std::string&                   xauthToken,
   const std::string&                   apiVersion,
-  bool                                 checkEntityExistance
+  const std::string&                   ngsiv2Flavour
```

Comment:
The idea is that flavours are not directly linked with request types, but with adjustable behaviours at mongoBackend, i.e. it could happen that the same operation could involve different flavour in the future or that two operations use the same flavour. Let's see how it evolves in the future but, by the moment, I prefer to have them as a separate thing.

Having said that, I don't like this flavour-based approach (in the same way I don't like the `apiVersion` branching in mongoBackend) but it is a "necessary evil" to survive without a complete re-work of the mongoBackend while NGSIv1 coexist with NGSIv2. In a situation in which NGSIv2 will be the only API version and the data types match the API, this is going to be removed.